api/app/models/job_file.rb

```diff
@@ -191,18 +182,32 @@ def <=>(other)
     filename <=> other.filename
   end
 
-  private
-
-  def payload_command
-    return @payload_command unless @payload_command.nil?
-    @payload_command = if @file_id == 'stdout'
-      FlightJobScriptAPI::JobCLI.view_job_stdout(@job_id, user: @user)
-    elsif @file_id == 'stderr'
-      FlightJobScriptAPI::JobCLI.view_job_stderr(@job_id, user: @user)
-    elsif decoded_file_id
-      FlightJobScriptAPI::JobCLI.view_job_results(@job_id, decoded_file_id, user: @user)
-    else
+  # Confirm the user has permission to access the file.
+  #
+  # We directly check the file permissions here to avoid launching a new
+  # `flight job` command
+  def is_readable?(id, path)
```

Comment:
This method doesn't need to take `id` and `path` as arguments. These are properties of `JobFile` and are otherwise static.

Reply:
Good spot.